common/app/common/InlineStyles.scala

```diff
@@ -93,21 +94,48 @@ object InlineStyles {
   }
 
   /**
+    * Execute some code n times, or until no exception is thrown
+    * (If n is 0, code still gets executed once)
+    */
+  def retry[T](n: Int)(r: => T)(implicit onFail: (Throwable, Int) => Unit) = {
+    def go(i: Int): Try[T] = {
+      Try(r) match {
+        case Failure(e) if i < n => {
+          onFail(e, i)
+          go(i + 1)
+        }
+        case f@Failure(e) => {
+          onFail(e, i)
+          f
+        }
+        case s@Success(v) => s
```

Comment:
If you aren't going to using `v` there is no point in capturing it. Could use underscore, or
`case s: Success[T] =>`